.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -379,7 +379,6 @@ jobs:
         shell: bash
         run: |
           cilium upgrade \
-            --helm-set=disableEnvoyVersionCheck=true \
```

Comment:
if we remove this one, might need to allow error log for version mismatch in cilium cli check error log.

Reply:
Why? I don't think we are expecting version mismatch to happen for prolonged time, no?

Reply:
Apologies, but I still don't get it :cry: 

What I can see is:
```
      - name: Upgrade Cilium
        if: ${{ matrix.skip-upgrade != 'true' }}
        shell: bash
        run: |
          cilium upgrade \
          ${{ steps.cilium-newest-config.outputs.config }} \
```
which should point to new helm chart:
```
      - name: Derive newest Cilium installation config
     [...]
          chart-dir: './untrusted/cilium-newest/install/kubernetes/cilium'
```
with upgraded Envoy?